### PR TITLE
Add quotes for environmental variables

### PR DESCRIPTION
**Pull request by weili11864**

### CODE REVIEW

File: service/handlers.go [hunk 1]
```diff
@@ -1267,8 +1267,11 @@ func generateEnvVariablesString(environmentVariables map[string]string) string {
 
 	r := "env"
 	for k, v := range environmentVariables {
-		tmp := r + fmt.Sprintf(" %s=%s", k, v)
-		r = tmp
+		if strings.HasPrefix(v, "'") {
```
Comment (jk-intuit) [inline service/handlers.go:1270]: we should take a different approach here.  

1. remove any single and double quotes
2. then surround the value in single quotes

things will get weird their are errant single quotes or anything mid string.

File: service/handlers_test.go [hunk 2]
```diff
@@ -842,7 +842,11 @@ func TestCreateWorkflowFromGit(t *testing.T) {
 				},
 			},
 			wfMock: &th.WorkflowMock{
-				SubmitFunc: func(ctx context.Context, from string, parameters, labels map[string]string) (string, error) {
+				SubmitFunc: func(ctx context.Context, from string, parameters map[string]string, labels map[string]string) (string, error) {
+					if parameters["environment_variables_string"] != "env foobar='barfoo'" {
```
Comment (jk-intuit) [inline service/handlers_test.go:846]: Lets add in more tests of that exercise different values in the environment variables. There are many combinations here that could trip up the quote replacement logic.

**Review by jk-intuit — CHANGES_REQUESTED** [review]

File: service/handlers.go (hunk 1 above)
Comment (weili11864) [reply service/handlers.go:1270]: quite not sure whether we need to remove any single or double quotes if single or double quotes are part of environmental variables.

Comment (weili11864) [reply service/handlers.go]: updated

**Review by jk-intuit — APPROVED** [review]

**Review by jk-intuit — APPROVED** [review]

**Pull request merged**